### PR TITLE
Update CI and adapt to 'dylan-tool'

### PR DESCRIPTION
**Pull request by fraya**
Adapt to 'dylan-tool':

- Remove old package information deft-package.json
- Remove registry/ folder
- Ignore development directories in .gitignore
- Change sphinx-extensions path to use dylan-tool's default

Documentation CI:

- Add a 'build-docs' action.
- Add version for epub documentation in 'conf.py'
- Use 'furo' theme in 'conf.py'
- Comment 'html_static_path' option in 'conf.py' to avoid warning building the docs.
- Ignore certification validation to avoid timeouts in 'linkcheck'

Build and test CI:

- Add dependabot
- Ignore build on changes in documentation
- Prepare build matrix for other OS.
- Update actions/checkout to v4
- Use dylan-tool to update and build

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -3,28 +3,43 @@ name: build-and-test
...
+    paths-ignore:
+      - 'documentation/**'
 
   # This enables the Run Workflow button on the Actions tab.
   workflow_dispatch:
 
 jobs:
   build-and-test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```
Comment (waywardmonkeys) [inline .github/workflows/build-and-test.yml:20]: Does this need to be a matrix if it is only running on a single platform and no other variation?

File: .github/workflows/build-docs.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  workflow_dispatch:
+
+# https://github.com/JamesIves/github-pages-deploy-action#readme
+permissions:
+    contents: write
+
+jobs:
+
+  build-and-deploy:
+    runs-on: ${{ matrix.os }}
```
Comment (waywardmonkeys) [inline .github/workflows/build-docs.yml:23]: Same question about needing a matrix.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (fraya) [reply .github/workflows/build-and-test.yml:20]: I leave it ready in case we want to add Mac OS later.

**Review by cgay — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (fraya) [reply .github/workflows/build-and-test.yml:20]: Should it be deleted?

Comment (cgay) [reply .github/workflows/build-and-test.yml:20]: It doesn't make much difference to me.  I guess in the long run it's good if we run most automation on both Linux and macOS.

Comment (fraya) [reply .github/workflows/build-and-test.yml:20]: Ok, then I merge.

**Pull request merged**